### PR TITLE
fix: add saved payment methods throughout checkout

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

The Saved Payment Methods screen was unable to navigate except via cards, but it should be accessible via all payment methods. I've relocated it from Card Payment to the Root component.

## How did you test it?

Go to every Payment Methods screen and check the - `Use saved payment method` button should be there to navigate to the Saved Payment Methods Screen.

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by prafulkoppalkar — APPROVED** [review]

**Review by vsrivatsa-juspay — DISMISSED** [review]

**Review by prafulkoppalkar — DISMISSED** [review]

File: src/Payments/CardPayment.res [hunk 1]
```diff
@@ -88,7 +88,7 @@ let make = (
     None
   }, (empty, complete))
 
-  let (savedMethods, isGuestCustomer) = React.useMemo1(() => {
+  let (_, isGuestCustomer) = React.useMemo1(() => {
```
Comment (ArushKapoorJuspay) [inline src/Payments/CardPayment.res:91]: If we are not going to use this savedMethods, please remove it from here

File: src/PaymentElement.res [hunk 2]
```diff
@@ -429,6 +430,23 @@ let make = (
         }}
       </div>
     </RenderIf>
+    <RenderIf
+      condition={displaySavedPaymentMethods && savedMethods->Array.length > 0 && showFields}>
```
Comment (ArushKapoorJuspay) [inline src/PaymentElement.res:434]: I think `showFields` already checks whether savedMethods list is empty or not. Could you please check this again

File: src/Payments/CardPayment.res (hunk 1 above)
Comment (ArushKapoorJuspay) [reply src/Payments/CardPayment.res:91]: Please remove this _ as well. useMemo should only return isGuestCustomer

File: src/PaymentElement.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/PaymentElement.res:434]: Not sure on this check. I think this array check should be there before it was also there. It's just a safe check to be there

File: src/Payments/CardPayment.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/Payments/CardPayment.res]: Updated

**Review by ArushKapoorJuspay — APPROVED** [review]

**Review by prafulkoppalkar — APPROVED** [review]

**Pull request merged**